File: ios/Sources/SIT/Views/Shared/WordmarkLockup.swift
import SwiftUI

/// Ticklr wordmark lockup — warm "Open Ink" theme.
///
/// Evolves the retired navy/amber card into an open, editorial lockup: the
/// Bebas Neue wordmark in warm ink sits straight on the surface, a hairline
/// rule above the "YOUR PEOPLE MATTER" tagline. No background — it adapts to
/// whatever paper/dark surface it's placed on, and flips ink colors for dark
/// mode automatically.
///
/// - `wordmarkSize`  point size of the Bebas wordmark (drives all other metrics)
/// - `showRule`      hairline divider; auto-suppress below ~36pt for legibility
/// - `dot`           optional terracotta full-stop ("Ticklr.") — one warm accent
struct WordmarkLockup: View {
    var wordmarkSize: CGFloat = 64
    var showRule: Bool = true
    var dot: Bool = false

    @Environment(\.colorScheme) private var scheme

    // ── Warm theme tokens ───────────────────────────────────────────────
    private var ink: Color {        // wordmark
        scheme == .dark ? Color(red: 0.957, green: 0.937, blue: 0.890)   // #F4EFE3
                        : Color(red: 0.102, green: 0.122, blue: 0.165)   // #1A1F2A
    }
    private var ink2: Color {       // tagline
        scheme == .dark ? Color(red: 0.722, green: 0.663, blue: 0.557)   // #B8A98E
                        : Color(red: 0.361, green: 0.392, blue: 0.439)   // #5C6470
    }
    private var ink3: Color {       // rule
        scheme == .dark ? Color(red: 0.486, green: 0.424, blue: 0.314)   // #7C6C50
                        : Color(red: 0.565, green: 0.600, blue: 0.643)   // #9099A4
    }
    private var terracotta: Color { // optional dot
        scheme == .dark ? Color(red: 0.816, green: 0.541, blue: 0.416)   // #D08A6A
                        : Color(red: 0.698, green: 0.388, blue: 0.259)   // #B26342
    }

    private var taglineSize: CGFloat { max(9, wordmarkSize * 0.17) }
    private var ruleVisible: Bool { showRule && wordmarkSize >= 36 }

    var body: some View {
        VStack(spacing: wordmarkSize * 0.16) {
            HStack(alignment: .bottom, spacing: wordmarkSize * 0.06) {
                Text("Ticklr")
                    .font(.custom("BebasNeue-Regular", size: wordmarkSize))
                    .tracking(-wordmarkSize * 0.012)
                    .foregroundStyle(ink)
                if dot {
                    Circle()
                        .fill(terracotta)
                        .frame(width: wordmarkSize * 0.15, height: wordmarkSize * 0.15)
                        .padding(.bottom, wordmarkSize * 0.10)
                }
            }
            if ruleVisible {
                Rectangle()
                    .fill(ink3.opacity(scheme == .dark ? 0.6 : 0.5))
                    .frame(width: wordmarkSize * 0.65, height: 1)
            }
            Text("YOUR PEOPLE MATTER")
                .font(.system(size: taglineSize, weight: .medium))
                .tracking(taglineSize * 0.34)
                .foregroundStyle(ink2)
        }
    }
}

#Preview("Light") {
    WordmarkLockup()
        .padding(40)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(Color(red: 1, green: 1, blue: 1))
        .preferredColorScheme(.light)
}

#Preview("Dark") {
    WordmarkLockup(dot: true)
        .padding(40)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(Color(red: 0.086, green: 0.075, blue: 0.063)) // #161310
        .preferredColorScheme(.dark)
}

File: android/app/src/main/java/com/xaymaca/sit/ui/shared/WordmarkLockup.kt
package com.xaymaca.sit.ui.shared

import androidx.compose.foundation.background
import com.xaymaca.sit.ui.theme.LocalIsAppDark
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import com.xaymaca.sit.ui.theme.BebasNeue

/**
 * Ticklr wordmark lockup — warm "Open Ink" theme.
 *
 * Replaces the retired navy/amber card with an open, editorial lockup: the
 * Bebas Neue wordmark in warm ink, a hairline rule above the
 * "YOUR PEOPLE MATTER" tagline. No background — adapts to its surface and
 * flips ink colors for dark mode.
 *
 * @param wordmarkSize size of the Bebas wordmark (drives all other metrics)
 * @param showRule     hairline divider; auto-suppressed below ~36sp
 * @param dot          optional terracotta full-stop ("Ticklr.") — one warm accent
 */
@Composable
fun WordmarkLockup(
    modifier: Modifier = Modifier,
    wordmarkSize: TextUnit = 64.sp,
    showRule: Boolean = true,
    dot: Boolean = false
) {
    val dark = LocalIsAppDark.current
    val ink        = if (dark) Color(0xFFF4EFE3) else Color(0xFF1A1F2A) // wordmark
    val ink2       = if (dark) Color(0xFFB8A98E) else Color(0xFF5C6470) // tagline
    val ink3       = if (dark) Color(0xFF7C6C50) else Color(0xFF9099A4) // rule
    val terracotta = if (dark) Color(0xFFD08A6A) else Color(0xFFB26342) // optional dot

    val sz = wordmarkSize.value
    val taglineSize = (sz * 0.17f).coerceAtLeast(9f).sp
    val ruleVisible = showRule && sz >= 36f

    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy((sz * 0.16f).dp)
    ) {
        Row(verticalAlignment = Alignment.Bottom) {
            Text(
                text = "Ticklr",
                style = TextStyle(
                    fontFamily = BebasNeue,
                    fontWeight = FontWeight.Normal,
                    fontSize = wordmarkSize,
                    lineHeight = wordmarkSize,
                    letterSpacing = (-0.012).em,
                    color = ink
                )
            )
            if (dot) {
                Spacer(Modifier.width((sz * 0.06f).dp))
                Box(
                    Modifier
                        .padding(bottom = (sz * 0.12f).dp)
                        .size((sz * 0.15f).dp)
                        .clip(CircleShape)
                        .background(terracotta)
                )
            }
        }
        if (ruleVisible) {
            Box(
                Modifier
                    .width((sz * 0.65f).dp)
                    .height(1.dp)
                    .background(ink3.copy(alpha = if (dark) 0.6f else 0.5f))
            )
        }
        Text(
            text = "YOUR PEOPLE MATTER",
            style = TextStyle(
                fontWeight = FontWeight.Medium,
                fontSize = taglineSize,
                letterSpacing = 0.34.em,
                color = ink2
            )
        )
    }
}
